assets/src/components/ladderPage.tsx

```diff
@@ -60,14 +64,17 @@ const LadderPage = (): ReactElement<HTMLDivElement> => {
       <RouteLadders
         routes={selectedRoutes}
         timepointsByRouteId={timepointsByRouteId}
+        vehiclesByRouteId={vehiclesByRouteId}
         selectedVehicleId={selectedVehicleId}
       />
 
       {selectedVehicleOrGhost && (
-        <PropertiesPanel
-          selectedVehicleOrGhost={selectedVehicleOrGhost}
-          route={vehicleRoute(routes, selectedVehicleOrGhost)}
-        />
+        <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
```

Comment:
Why not just wrap everything here with this and not have to change RouteLadders?

Reply:
If it's just going one level deep, normal arguments are better than contexts. Contexts are good for avoiding prop drilling, but that's for the benefit of the intermediate components that would do nothing but forward the argument along. For the consumer, RouteLadders, getting the value from an argument is more direct and less magic than getting it from a context.

Reply:
I can see how if you are only passing a piece of data into a single component why you wouldn’t need the overhead of a context, but once you already have a context it feels like consistency becomes much more important. For the consumer it’s much simpler to have one way of accessing this data than to think about different choices each time.